src/encoded/commands/deploy.py

```diff
@@ -232,7 +230,7 @@ def _get_bdm(main_args):
 def get_user_data(commit, config_file, data_insert, main_args):
     cmd_list = ['git', 'show', commit + config_file]
     config_template = subprocess.check_output(cmd_list).decode('utf-8')
-    ssh_pub_key = read_ssh_key()
+    ssh_pub_key = read_ssh_key(main_args.identity_file)
```

Comment:
I have not ran this myself so this may be an odd question. The variable- main_args is set in main(). However, it seems internal to that method. It may be wise to push it up to the class-level or __init__, so reads know it is class-wide.

Reply:
It has to be instantiated in main to get the command line args.  Ideally this script would subclass most of what is in main(), then we instantiate some "TBD" DeployClass where the main_args would be the initialization arguments.

yea, handing main_args though all the functions is a little weird.

Reply:
I got the answer. main() passes main-args around. It is safe to assume main() is called prior to any other object getting called.